File: sdks/node/src/api/plugin/mapper.ts
import type { PluginRes } from "./res.js";
import type { PluginOutput } from "../../domain/plugin/output.js";
import type { PluginReq } from "./req.js";
import type { PluginInput } from "../../domain/plugin/input.js";

class PluginMapper {
  public static ToDomain(req: PluginReq): PluginInput {
    return {
      directory: req.directory,
      config: req.config,
    } satisfies PluginInput;
  }

  public static ToRes(res: PluginOutput): PluginRes {
    return {
      outputDir: res.directory,
    };
  }
}

export { PluginMapper };

File: sdks/python/read_dir/plugin/mapper.py
from api.plugin.req import PluginReq
from api.plugin.res import PluginRes
from domain.plugin.input import PluginInput
from domain.plugin.output import PluginOutput


class PluginMapper:
    @staticmethod
    def to_domain(req: PluginReq) -> PluginInput:
        return PluginInput(directory=req.directory, config=req.config)

    @staticmethod
    def to_res(res: PluginOutput) -> PluginRes:
        return PluginRes(output_dir=res.directory)
